assets/src/hooks/useDetours.ts

```diff
@@ -0,0 +1,244 @@
+import { Channel, Socket } from "phoenix"
+import { SimpleDetour } from "../models/detoursList"
+import { useEffect, useState } from "react"
+import { reload } from "../models/browser"
+import { userUuid } from "../util/userUuid"
+import { RouteId } from "../schedule"
+
+interface DetoursMap {
+  [key: number]: SimpleDetour
+}
+
+const subscribe = (
+  socket: Socket,
+  topic: string,
+  initializeChannel: React.Dispatch<React.SetStateAction<DetoursMap>>,
+  handleDrafted: ((data: SimpleDetour) => void) | undefined,
+  handleActivated: ((data: SimpleDetour) => void) | undefined,
+  handleDeactivated: ((data: SimpleDetour) => void) | undefined
+): Channel => {
+  const channel = socket.channel(topic)
+
+  handleDrafted &&
+    channel.on("drafted", ({ data: data }) => handleDrafted(data))
+  handleActivated &&
+    channel.on("activated", ({ data: data }) => handleActivated(data))
+  handleDeactivated &&
+    channel.on("deactivated", ({ data: data }) => handleDeactivated(data))
+  channel.on("auth_expired", reload)
+
+  channel
+    .join()
+    .receive("ok", ({ data: data }: { data: SimpleDetour[] }) => {
+      const detoursMap = data.reduce(
+        (acc, detour) => ({ ...acc, [detour.id]: detour }),
+        {}
+      )
```

Comment:
suggestion(non-blocking): small syntax thing that I think makes it easier to read
```suggestion
      const detoursMap = Object.fromEntries(data.map(v => [v.id, v]))
```